deepchem/models/keras_model.py

```diff
@@ -969,14 +971,89 @@ def restore(self, checkpoint=None, model_dir=None):
     if tf.executing_eagerly():
       self._checkpoint.restore(checkpoint)
     else:
-      self._checkpoint.restore(checkpoint).run_restore_ops(self.session)
+      if session is None:
+        session = self.session
+      self._checkpoint.restore(checkpoint).run_restore_ops(session)
 
   def get_global_step(self):
     """Get the number of steps of fitting that have been performed."""
     if tf.executing_eagerly():
       return int(self._global_step)
     return self._global_step.eval(session=self.session)
 
+  def default_assignment_map(self, source_model, include_top=True, **kwargs):
```

Comment:
How about making this method private?  That is, rename it to `_default_assignment_map()`.  It's just there to be called from `load_from_pretrained()`, not for users to call directly.

Reply:
While the current API does not reflect that, with the `default_assignment_map`, I was aiming for something like `default_generator`. But I am not sure if it is better for someone to change `default_assignment_map` or supply an `assignment_map`. I think the latter is the preferred option and in that setting it should be made private.

Reply:
This is done.